msgpack/shared/src/test/scala/fs2/data/msgpack/ParserSpec.scala

```diff
@@ -0,0 +1,274 @@
+      .map(expect(_))
+      .compile
+      .foldMonoid
+  }
+
+  test("MessagePack value parser should raise when invalid timestamp length is used") {
+    val cases = List(
+      // fixext
+      hex"0xd4ff00", // length 1
+      hex"0xd5ff0000", // length 2
+
+      // ext
+      hex"0xc701ff00", // length 1
+      hex"0xc70dff00000000000000000000000000" // length 13
+    )
+
+    Stream
+      .emits(cases)
+      .evalMap { n =>
+        Stream
+          .chunk(Chunk.byteVector(n))
+          .through(low.items[IO])
+          .compile
+          .toList
+          .attempt
+          .map(_.isLeft)
+          .map(expect(_))
+      }
+      .compile
+      .foldMonoid
+  }
+  test("MessagePack value parser should correctly parse example formats") {
+    val cases = List(
+      // [ 1, 2, 3 ]
+      (hex"93 01 02 03",
+       List(MsgpackItem.Array(3),
+            MsgpackItem.SignedInt(hex"01"),
+            MsgpackItem.SignedInt(hex"02"),
+            MsgpackItem.SignedInt(hex"03"))),
+
+      /* [
+       *   "on",
+       *   {
+       *     "title": "Window",
+       *     "name": "main_window",
+       *     "width": 500,
+       *     "height": 500
+       *   }
+       * ]
+       */
+      (hex"92 A2 6F 6E 84 A5 74 69 74 6C 65 A6 57 69 6E 64 6F 77 A4 6E 61 6D 65 AB 6D 61 69 6E 5F 77 69 6E 64 6F 77 A5 77 69 64 74 68 CD 01 F4 A6 68 65 69 67 68 74 CD 01 F4",
+       List(
+         MsgpackItem.Array(2),
+         MsgpackItem.Str(hex"6f6e"),
+         MsgpackItem.Map(4),
+         MsgpackItem.Str(hex"7469746C65"),
+         MsgpackItem.Str(hex"57696E646F77"),
+         MsgpackItem.Str(hex"6E616D65"),
+         MsgpackItem.Str(hex"6D61696E5F77696E646F77"),
+         MsgpackItem.Str(hex"7769647468"),
+         MsgpackItem.UnsignedInt(hex"1f4"),
+         MsgpackItem.Str(hex"686569676874"),
+         MsgpackItem.UnsignedInt(hex"1f4")
+       )),
+
+      /* {
+       *    "int": 1,
+       *    "float": 0.5,
+       *    "boolean": true,
+       *    "null": null,
+       *    "string": "foo bar",
+       *    "array": [
+       *      "foo",
+       *      "bar"
+       *    ],
+       *    "object": {
+       *      "foo": 1,
+       *      "baz": 0.5
+       *    },
+       *    "timestamp": <9:02:47 pm UTC + 280716ns, August 20, 2414>
+       * }
+       */
+      (hex"""
+        88
+        A3 69 6E 74
+        01
+        A5 66 6C 6F 61 74
+        CA 3F 00 00 00
+        A7 62 6F 6F 6C 65 61 6E
+        C3
+        A4 6E 75 6C 6C
+        C0
+        A6 73 74 72 69 6E 67
+        A7 66 6F 6F 20 62 61 72
+        A5 61 72 72 61 79
+        92
+        A3 66 6F 6F
+        A3 62 61 72
+        A6 6F 62 6A 65 63 74
+        82
+        A3 66 6F 6F
+        01
+        A3 62 61 7A
+        CB 3F E0 00 00 00 00 00 00
+        A9 74 69 6D 65 73 74 61 6D 70
+        D7 FF 00 11 22 33 44 55 66 77
+        """,
+       List(
+         MsgpackItem.Map(8),
+         MsgpackItem.Str(ByteVector.encodeUtf8("int").toOption.get),
```

Comment:
I am a bit bothered by these `toOption.get` for some reason. We know it will not fail here, so we can avoid wrapping by using this:

```suggestion
         MsgpackItem.Str(ByteVector("int".getBytes(StandardCharsets.UTF_8))),
```

Reply:
Agreed, this approach is a cleaner and makes Codacy happy.